### PR TITLE
Compatibility updates

### PR DESCRIPTION
**Pull request by zikani03**
Includes updates for compatibility with php 5.4, support for non-conventional primary keys and a utility trait for getting audit sources

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,105 +4,132 @@ A logging plugin for [CakePHP](http://cakephp.org). The included `AuditableBehav
 
 The behavior tracks changes on two levels. It takes a snapshot of the fully hydrated object _after_ a change is complete and it also records each individual change in the case of an update action.
 
+Based on the Original AuditLog plugin for CakePHP 2.x found [here](https://github.com/robwilkerson/CakePHP-Audit-Log-Plugin) and off the awesome work of [rochamarcelo](https://github.com/rochamarcelo)
```
Comment (jippi) [inline README.md]: seems irrelevant here, I don't mind attribution, but move it to the footer of the file you want it there - almost all of it has been rewritten multiple times since the original fork :) it's also visible in the "forked from" header for the repo - so a bit redundant

**Comment by jippi** [comment]
should get it moved to FoC, don't you think @lorenzo  ?

**Comment by zikani03** [comment]
Updated the read me, waiting for your response guys :)

**Comment by jippi** [comment]
We (Bownty) are moving to this full reimplementation - https://github.com/lorenzo/audit-stash - pending move to FoC when @lorenzo got time to do so :) 

As such, this plugin will not support cakephp 3.x, but will remain compatible for cakephp 2.x only
